internal/nodes/avi_model_evh_nodes.go

```diff
@@ -2099,6 +2099,12 @@ func RouteIngrDeletePoolsByHostnameForEvh(routeIgrObj RouteIngressModel, namespa
 		}
 
 		SharedHostNameLister().DeleteNamespace(host)
+		if found, ingressHostMap := SharedHostNameLister().Get(host); found {
+			mapkey := namespace + "/" + objname
+			if utils.HasElem(ingressHostMap.GetIngressesForHostName(), mapkey) {
+				delete(ingressHostMap.HostNameMap, mapkey)
+			}
```

Comment:
Once value is delete from map. We should save the updated map for that host.

Reply:
this update to the map should be in-place as maps are passed by reference

Reply:
Do we need to guard delete of map with if condition?

Reply:
not required, removed the if condition